### PR TITLE
Rename API to points and rewards

### PR DESCRIPTION
**Pull request by ethanfrey**
Part 1 of https://github.com/confio/tgrade-contracts/issues/272 
(the rest must be done in tgrade-contracts after merging this, tagging and updating deps).

Become more consistent in naming.

**points** are used for non-transferable items like voting power

**rewards** are used for funds being paid out

I believe I made all needed adjustments in `msg.rs` and `state.rs` files, aka the public interfaces. Some docstrings need to be updated, but more importantly, a whole lot of local variables.

I just made the first pass to clarify which names are desired where. After 0.5.5 is tagged, someone can please take this branch over (make a new PR if you want) 

TODO:
- [ ] Check over local variable and function names
- [ ] Replace weight -> points in usages of tg3::Vote/VoteInfo (all voting contracts) - do not change in cw-plus, but rather use a custom version of those (tg3?) in our packages with consistent naming
- [x] Add a MIGRATION.md that has a brief overview of breaking changes in public API (to help Go and JS clients update quickly)

### CODE REVIEW

**Comment by ethanfrey** [comment]
@hashedone are you picking this up? If not, can you hand it to someone else?

It's been sitting here for a while and needs rebase and more.

**Comment by hashedone** [comment]
So the rebase is done, but I just grepped for the `weight` - there is no world I would include this in the same PR, it is to much for one review. As those changes are not breaking, I would:
1. Add info about changes to MIGRATION.md
2. Try to merge this as it is
3. Do " Replace weight -> points in usages of tg3::Vote/VoteInfo (all voting contracts) - do not change in cw-plus, but rather use a custom version of those (tg3?) in our packages with consistent naming" in separate PR
4. Do the local variables change in separate PR

**Comment by hashedone** [comment]
I would really like to close it as it is to not create monstrum PR. It covers most API breaking changes, there  would be small follow up for cw3 changes (creating tg3 equivalent), and only then I would go with cleaning up local symbols which should be non breaking.

**Comment by ethanfrey** [comment]
> Add info about changes to MIGRATION.md
> Try to merge this as it is

Sounds good. I will review now and merge if approved

> Do " Replace weight -> points in usages of tg3::Vote/VoteInfo (all voting contracts) - do not change in cw-plus, but rather use a custom version of those (tg3?) in our packages with consistent naming" in separate PR

Sounds good, this also needs MIGRATION entry obviously

> Do the local variables change in separate PR

Fine to do that last. That should nicely not conflict with anything else, and even roll out in a patch release.

Reviewing now

File: MIGRATING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Migrating
+
+This guide lists API changes between releases of *PoE* contracts.
+
+## 0.6.0-beta1 -> 0.6.0-beta2
```
Comment (ethanfrey) [inline MIGRATING.md:5]: Nice guide. Thanks.

Minor point: I would label the next step 0.6.0-rc1 (with all the breaking api changes), not beta2

File: contracts/tg4-engagement/src/multitest/suite.rs [hunk 2]
```diff
@@ -45,7 +45,7 @@ impl SuiteBuilder {
     pub fn with_member(mut self, addr: &str, weight: u64) -> Self {
         self.members.push(Member {
             addr: addr.to_owned(),
-            weight,
+            points: weight,
```
Comment (ethanfrey) [inline contracts/tg4-engagement/src/multitest/suite.rs:48]: Was going to point this out, but yeah, fine to handle variables in the follow up PR

File: contracts/tgrade-valset/src/multitest/update_config.rs [hunk 3]
```diff
@@ -8,39 +8,39 @@ use super::suite::SuiteBuilder;
 fn update_cfg() {
     let mut suite = SuiteBuilder::new()
         .with_max_validators(6)
-        .with_min_weight(3)
+        .with_min_points(3)
```
Comment (ethanfrey) [inline contracts/tgrade-valset/src/multitest/update_config.rs:11]: nice to get this too

File: packages/tg4/src/helpers.rs [hunk 4]
```diff
@@ -194,7 +194,7 @@ impl Tg4Contract {
         limit: Option<u32>,
     ) -> StdResult<Vec<Member>> {
         let query =
-            self.encode_smart_query(Tg4QueryMsg::ListMembersByWeight { start_after, limit })?;
+            self.encode_smart_query(Tg4QueryMsg::ListMembersByPoints { start_after, limit })?;
```
Comment (ethanfrey) [inline packages/tg4/src/helpers.rs:197]: The method name above is still `list_members_by_weight`, please update

**Review by ethanfrey — COMMENTED** [review]
Looks good.
2 minor changes requested.

Please just push those and merge.

**Review by hashedone — APPROVED** [review]
LGTM

**Pull request merged**